File: src/main/kotlin/com/developersam/webcore/service/GoogleUser.kt
package com.developersam.webcore.service

import com.google.firebase.auth.FirebaseToken

/**
 * A [GoogleUser] is an class that contains many information of a Google user,
 * obtained from a token.
 */
class GoogleUser internal constructor(token: FirebaseToken) {

    /**
     * UID of the user.
     */
    val uid: String = token.uid
    /**
     * Name of the user.
     */
    val name: String = token.name
    /**
     * Email of the user.
     */
    val email: String = token.email
    /**
     * Picture of the user.
     */
    val picture: String = token.picture

}

File: frontend/src/app/shared/google-user.ts
/**
 * [GoogleUser] represents a [GoogleUser] in the system.
 */
export interface GoogleUser {
  /**
   * Used to uniquely identifiers a user.
   */
  readonly key: string;
  /**
   * Name of the user.
   */
  readonly name: string;
  /**
   * Email of the user.
   */
  readonly email: string;
  /**
   * Picture url of the user.
   */
  readonly picture: string;
}
